src/server/game/Server/Packets/MiscPackets.cpp

```diff
@@ -711,3 +711,32 @@ WorldPacket const* WorldPackets::Misc::StartTimer::Write()
 
     return &_worldPacket;
 }
+
+void WorldPackets::Misc::AdventureJournalOpenQuest::Read()
+{
+    _worldPacket >> AdventureJournalID;
+}
+
+void WorldPackets::Misc::AdventureJournalStartQuest::Read()
+{
+    _worldPacket >> QuestID;
+}
+
+void WorldPackets::Misc::AdventureJournalUpdateSuggestions::Read()
+{
+    OnLevelUp = _worldPacket.ReadBit();
+}
+
+WorldPacket const* WorldPackets::Misc::AdventureJournalDataResponse::Write()
+{
+    _worldPacket.WriteBit(OnLevelUp);
+    _worldPacket.FlushBits();
+    _worldPacket << uint32(AdventureJournalDatas.size());
+    for (auto const& dataInfo : AdventureJournalDatas)
+    {
+        _worldPacket << dataInfo.AdventureJournalID;
```

Comment:
uses casts like you did with .size()

e.g.
```c
_worldPacket << uint32(dataInfo.AdventureJournalID);
```
in this case

do this for all primitive datatypes